2-mandatory/3-shopping-list.js

```diff
@@ -20,6 +20,15 @@ let pantry = {
 
 function createShoppingList(recipe) {
   // write code here
+  let obj = {};
+  obj.name = recipe.name;
+  recipe.ingredients.forEach((item) => {
+    if (!pantry.fridgeContents.concat(pantry.cupboardContents).includes(item)) {
+      obj.items = obj.items || [];
```

Comment:
greate job, here you do not need this line obj.items = obj.items || [] , could you explain to me why you wrote this line?

Reply:
this line will check if the property exists it will just be equal to itself if not will create it with a value of an empty array